### ISSUE TITLE
Autodoc Installations are too expensive and result in boring gameplay

### ISSUE THREAD

**Issue by John-Candlebury** [CleverRaven/Cataclysm-DDA#24376]
The new method for installing bionics is too expensive, especially the anesthesia requirement, and is detrimental to fun game.  It relegates bionics to the far late game, or even worse, makes a versatile and interesting cyborg build completely unachievable due to the long timescale required to loot anesthetic. 

The looting also produces no memorable or particularly interesting challenges, it consists mostly of driving around the overmap hoping for favorable loot rolls.

The problem is particularly evident when it comes to obtaining power storage.

Some suggestions:
-Allow a single use of anesthetic to install/uninstall more than a single CBM.
-Guaranteed spawns for some small amount of anesthetic in every Auto doc Spawn.
-Buff the capacity of power storage bionics.
-More common spawns for anesthesia, preferably in specialized locations, that are more interesting to loot.
-Give charges to the anesthesia kit.

**Comment by Night-Pryanik** [comment]
And who said installing bionics should be a) easy, and b) an early-game thing?

**Comment by John-Candlebury** [comment]
>And who said installing bionics should be a) easy, and b) an early-game thing?

No one, the problem is that now the timescale required to get bionics is so long to be completely detached from  the game difficulty scaling, this makes them mostly irrelevant from a game play point of view, especially for bionics meant for combat. 

Once you have a decent bionic load out, and the power to use those, you have completely survived and most likely out-scaled every difficult enemy in-game without the bionics. After all it certainly took you several seasons of driving around to find enough anesthesia kits, all done while fighting hulks and predators. . So any advantage provided from bionics are minimal or just aQol improvements.

The other problem is that several seasons of  anesthetic looting is boring but not harder or more memorable than usual game play.

Basically the problem with the current system is that, once you can have them, you have little real need for them.

**Comment by Night-Pryanik** [comment]
> Basically the problem with the current system is that, once you can have them, you have little real need for them.

That's the "problem" not of autodoc or anesthesia kit, but of difficulty of bionic installation. Early-game characters in most cases just don't have skills high enough to reliably install bionics. Pre-autodoc system required high skills too.

**Comment by 0be1isk** [comment]
I agree, anesthesia kits should most definitely have charges
Anesthesia is generally used in two ways, through a syringe or through a gas tank. There should be no reason why it's a one time use thing.

**Comment by AMurkin** [comment]
I think CBMs must have tiers. And higher tier needs more time to install. And anesthesia kit should provide some time of the effect. Than player choose CBMs he wants to install within that time.
And probably actual time of installation must vary randomly.

**Comment by John-Candlebury** [comment]
The time problem is strictly from having to scavenge anesthesia kits from  rare buildings separated by very long distances.  the lack of anesthesia takes much longer to resolve thanfinding an autodoc, bionics, and getting the skills necessary for instalation.

You can get the skills bionics and find an autodoc in less than two weeks, but depending on your luck,  mods a finding 3 charges of anaesthesia (minimum for a power source, storage +active bionic) could take you disproportionally longer.

**Comment by Xhuis** [comment]
If the current system remains, bionics should be buffed to compensate for the massively increased barrier towards installing them. Nobody in their right mind would ever want to use a full anesthetic kit to install a bionic clock, for instance.

While I'm overall supportive of the change because so much thought was put into making it as smooth as possible, in its current state it's quite rough around the edges. I tried to address it a little in #24368, but it doesn't fully solve the problem.

The simplest solution in my eyes is to modify installation time based on difficulty - say, minutes equal to `20 * difficulty` - and make anesthesia kits into charge-items with, say, 4 hours of anesthesia in them; that way, super basic things like the Chronometer, with a difficulty of 1, aren't cripplingly consuming to install compared to the benefit they grant, and very powerful ones like the forcefields are more impactful due to their complexity and strength.

Anesthesia kits themselves would have charges deducted to place the player into anesthesia; when the operation begins for a difficulty 3 bionic, it would lose the amount of charges equal to an hour, and place the player into narcosis for that hour.

Would this be an alright solution?

**Comment by tinukedaya** [comment]
What if the anesthesia would simply be part of the Autodoc process and not extra item... it already is forcing you to clear the area around to make sure you won't be killed during the sleep... And if needed for balance reason after you wake up give some hard dizzy debuff will slowly go away thru the next 24 hours. 

If you still want to keep the extra requirement, lets give it limited amount of anesthesia gas it to support just a few operations. then you would need to refill it. and instead of the anesthesia kit you can find tanks with the gas in some places. One tank can again last for few operations. Maybe even small chance for ambulance. 

It is basically the same as proposed above with the kit having charges and guaranteed drops with autodoc, but somehow it feels more real for me.  

And you can have different amount of anesthesia used for different CBM's varying the time you're "gone" based on the surgery complexity.

**Comment by kevingranade** [comment]
> Allow a single use of anesthetic to install/uninstall more than a single CBM.
> Give charges to the anesthesia kit.

These are the same thing, effectively "spawn more anesthesia at a time".  I'm not against it.
I agree with the suggestion to have the kit put the player under for a certain amount of time, and allow multiple installations to occur in that time, with different CBMs requiring different amounts of time.

> Guaranteed spawns for some small amount of anesthetic in every Auto doc Spawn.

Against this kind of thing in general, was the area looted? Have supplies been destroyed? Overly convenient guaranteed spawns have a very gamey feel to them.

> Buff the capacity of power storage bionics.

I think you're going to need to open another issue for this as it's mostly unrelated.
> More common spawns for anesthesia, preferably in specialized locations, that are more interesting to loot.

This is my favorite suggestion, we can certainly have more hospitals and clinics, and the spawn rate can also be reexamined.  Anywhere that surgery might happen would have a decent chance of anesthesia spawning.  This includes vet offices and potentially facilities where animals are kept.

> If the current system remains, bionics should be buffed to compensate for the massively increased barrier towards installing them.

That doesnt follow, bionics have historically been way too easy to acquire, and many of them are already game-breakingly powerful.

> Nobody in their right mind would ever want to use a full anesthetic kit to install a bionic clock, for instance.

So don't, theres nothing wrong with that.

> To isolate nitrogen, all you need to do is...

Two options, rather similar in apparent very high crafting difficulty.  Find a "how to make a nitrogen generator at home" page and we can talk.  I'm not saying it's not possible, but your sources make no case for it being a simple operation.

You also have no sources for creation of NO from N2, just a bare assertion that it's easy.

**Comment by tinukedaya** [comment]
>You also have no sources for creation of NO from N2, just a bare assertion that it's easy.

Found one that seems to be really quite easy... We would need Chemistry set, fertilizer, sand and water. Cooking 4-5?
https://fear-of-lightning.wonderhowto.com/news/make-your-own-nitrous-oxide-hilarious-science-0134142/

**Comment by Lasiace** [comment]
Just piping in with regards to nitrous oxide, while I'm not familiar with the process that Amariithynar talks about, there are two ways that I think could be implemented. I am basing this on [this wikipedia article.](https://en.wikipedia.org/wiki/Nitrous_oxide#Production)

The first is through the use of ammonium nitrate, which is already modeled in-game. If you _carefully_ heat it to around 250 C, it will naturally decompose into nitrous oxide and water vapour through this chemical reaction:

### NH<sub>4</sub>NO<sub>3</sub>  → 2 H<sub>2</sub>O + N<sub>2</sub>O

This is how they typically do it industrially. The main issue with this approach is that if you _don't_ do this process carefully, the concoction _explodes._ I don't think that crafting can result in explosions yet, however, so I don't know how you'd handle the consequences of failing this recipe.

An alternative, safer method the article outlines is the use of urea, nitric acid, and sulfuric acid. Here is the chemical reaction for that one:

### 2 (NH<sub>2</sub>)<sub>2</sub>CO + 2 HNO<sub>3</sub>+ H<sub>2</sub>SO<sub>4</sub> → 2 N<sub>2</sub>O + 2 CO<sub>2</sub> + (NH<sub>4</sub>)<sub>2</sub>SO<sub>4</sub> + 2H<sub>2</sub>O.

Of these, two are already in-game, while urea is not. However, [urea could be made using ammonia and carbon dioxide](https://en.wikipedia.org/wiki/Urea#Production), and carbon dioxide can be made in large quantities by burning charcoal, so perhaps we can use those items in the recipe instead, along with some heating charges for the burning.

As for the preparation of this, you would just need to add all of these and mix them carefully in a flask (so basically a chem set in-game), starting with the sulfuric acid, then adding in the urea powder, and slowly adding in the nitric acid. The whole process takes 5 hours, and needs to be at about 70 C, so additional heating charges would be needed here. Afterwards, you clean the resulting mixture with sodium hydroxide (that is, lye) and you'll have some nitrous oxide. I'm using the [actual source](https://www.erowid.org/archive/rhodium/chemistry/nitrous.html) instead of wikipedia for this one, see Example 1 in that article.

This process requires significantly more things to pull off (sulfuric acid, nitric acid, ammonia, charcoal, and lye), but that's simply how it goes. Perhaps balance it out by requiring less skill, since you wouldn't need to be as careful, unlike the ammonium nitrate recipe.

I don't really talk about the other reactions the article outlines since those use a lot of chemicals that aren't implemented in-game yet.

Overall, with these processes it certainly seems possible. Not easy, but possible, and the game already has the ingredients needed for the reaction, so it would just be a few extra recipes implementation-wise. Well, unless you want to add exploding recipes.

**Comment by Amariithynar** [comment]
Urea is also easily obtained from refining urine (from which it gets its name), though we don't actually produce urine in-game; it's a very common component of any mammalian life; just need to treat it with alcohol through successive filtrations and evaporations to clean it of any impurities, as it's not ether soluble. 

Even if we discard that entirely, using a hotplate for consistant temperature (as they are dial-based and have an internal temperature sensor), or for added security with a thermometer, would allow relatively stable production utilizing the decomposition method.

@kevingranade Nitrogen gas reacts with oxygen under electric current or extremely high temperatures (too high for our purposes) to create nitrogen monoxide https://melscience.com/en/articles/nitrogen-and-oxygen-and-their-interaction/, which reacts spontaneously with oxygen to create nitrogen dioxide http://apps.sepa.org.uk/spripa/pages/substanceinformation.aspx?pid=122, which when bubbled through water produces nitric acid http://pieceofscience.com/nitric-acid-preparation/, which when combined with ammonia creates ammonium nitrate https://pubchem.ncbi.nlm.nih.gov/compound/ammonium_nitrate#section=Methods-of-Manufacturing, which can be decomposed at high heat (between 170-240 degrees celcius; higher risks explosive decomposition) to create nitrous oxide (N2O) and water vapor (H2O) http://www.societechimiquedefrance.fr/extras/Guiochon%20VO/decompositionVO.htm; you can also stop before you react the ammonium nitrate step and add iron filings to nitric acid and capture the gas that bubbles off from that, which is also nitrous oxide. https://toxnet.nlm.nih.gov/cgi-bin/sis/search2/f?./temp/~7bkfzR:1:mmfg https://www.thoughtco.com/make-nitrous-oxide-or-laughing-gas-608280

**Comment by Xhuis** [comment]
~~All this talk about anesthesia when you could just be huffing paint out of a sock anyway~~

**Comment by Night-Pryanik** [comment]
> The looting also produces no memorable or particularly interesting challenges, it consists mostly of driving around the overmap hoping for favorable loot rolls.

This can be said to the whole CBM obtaining process. We're driving around, hoping for favorable spawn rolls i.e. for finding zombie technicians, shockers or bio-ops to kill and then butcher them. No memorable or particularly interesting challenge.

**Comment by Amariithynar** [comment]
>This can be said about the whole CBM obtaining process. We're driving around, hoping for favorable spawn rolls i.e. for finding zombie technicians, shockers or bio-ops to kill and then butcher them. No memorable or particularly interesting challenge.

Perhaps make it so that you can only ever cut out burnt out bionics and components that are part of the recipe for the bionic, but then those can be crafted (assuming you know the recipe, obtained from books only) into other CBMs, which you can then install at an autodoc? or maybe "damaged XXX CBM" that needs to be disassembled/repaired before it can be installed, like how we can disassemble broken manhacks to then assemble them into inactive manhacks on our side? This would put it more in line with Mutagen, basically, but would still be the primarily active 'mutation' route, compared to Mutagen's primarily passive 'mutation' route, with a few notable CBMs, like the Integrated Toolset.

Either way, the anesthesia is still far too uncommon for what is really a common/easy to produce product, and that needs to be addressed.

**Comment by Firestorm01X2** [comment]
Looks like simpliest solution:
Adding more anasthesia kits to spawn AND adding recepie to craft anasthesia.

It should work.

**Comment by Night-Pryanik** [comment]
> And why no one find hillarious that anasthesia is more rare than CBMs itself?

Remove the ability to get install-ready CBMs from the bodies (it doesn't matter if it is completely removed, or if all salvaged CBMs should be filthy/damaged/whatelse, or some other option) and SUDDENLY CBMs become ultra-rare.

**Comment by Firestorm01X2** [comment]
Thats is exactly could replace anatsthesia kit dancing.
But it is kinda too late.

**Comment by Amariithynar** [comment]
Regardless of CBM rarity shooting through the roof, anaesthesia is still far too rare in-game when it's quite common. As such, there's basically two issues at hand here that both need resolution: More accessible anaesthesia, though it should require a fair amount of chemistry AND first aid knowledge, and a full chemistry kit/lab, to produce; and rarer CBMs in general, that aren't as easy to obtain as just cutting them out of a Zombie. Personally, I think Power Storage CBMs should be basically fine to be able to remove, as they're effectively just implanted rechargable batteries, but the rest should disassemble into their components (which if you're high enough on skill, should grant the chance to learn the recipe to put together the components needed for the autodoc to install them).

**Comment by Night-Pryanik** [comment]
> Now let's just add more kits.

Feel free to buff anesthesia spawn as you see fit.

> More accessible anaesthesia, though it should require a fair amount of chemistry AND first aid knowledge, and a full chemistry kit/lab, to produce

Feel free to open a PR adding recipe for this.

**Comment by Amariithynar** [comment]
>Feel free to open a PR adding recipe for this.
I would if I knew the least amount of how to code, especially for CDDA.

**Comment by DracoGriffin** [comment]
>I would if I knew the least amount of how to code, especially for CDDA.

Adding recipes requires json, which is extremely easy to modify/"code". I'd suggest taking a look at item jsons to get an idea of the end-result, and look at how recipe jsons on how it happens.

[Knife spear](https://github.com/CleverRaven/Cataclysm-DDA/blob/20ab49b2413ea9f5bfc529835f754b7671e7a172/data/json/items/tools.json#L6715) is a [good example](https://github.com/CleverRaven/Cataclysm-DDA/blob/20ab49b2413ea9f5bfc529835f754b7671e7a172/data/json/recipes/recipe_weapon.json#L261) to start with.

If you're still having a hard time, make sure to check the doc folders, mainly for [JSON_FLAGS.md](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/JSON_FLAGS.md) and [JSON_INFO.md](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/JSON_INFO.md).

If you still can't figure it out, well... I'd chalk it up to laziness rather than ability, as your comments and issues you've posted have demonstrated a level of intelligence that should be quite capable of learning json modification.

**Comment by Amariithynar** [comment]
Since I'm wanting to do a chemical, going to use https://github.com/CleverRaven/Cataclysm-DDA/blob/20ab49b2413ea9f5bfc529835f754b7671e7a172/data/json/recipes/recipe_medsandchemicals.json and https://github.com/CleverRaven/Cataclysm-DDA/blob/20ab49b2413ea9f5bfc529835f754b7671e7a172/data/json/items/chemicals_and_resources.json for resources, but as I still have no idea how to Git besides very basic "hit button to create new issue" functionality, I'll need to hit up the Discord. Thanks for both the compliment and assistance on that point, @DracoGriffin.

**Comment by Night-Pryanik** [comment]
The only thing is utterly broken here is your logic. You are playing demigod, okay. So what? Anesthesia kits should start raining down on you like from cornucopia? Have you ever thought that there may be some items that should spawn rarely? Maybe not so rare as they are now (this is being fixed now, by the way, in case you didn't know), but rare nevertheless, no matter how cool you think your character is.

Autodoc was designed in a way that it needs to be configured prior to every operation. It just executes the program skilled operator entered in it. And the fact that you're thinking that autodoc is a some sort of an AI that should always have 100% chance of success is entirely your problem. By the way, in case you didn't know, there is a PR ready to be merged, so you can make autodoc conduct operations the way YOU think it should.

Rarely any complex feature is implemented perfect by the very first PR. Reality is not like your wet fantasies in which game-changing features are licked to shine by a single PR. Almost always there are bugs, errors, and the need for fine-tuning. Too bad the ones like you don't understand simple things: this is experimental version of the game, the feature are constantly being fixed/enhanced/updated based on constructive feedback and not shitty whining like "This system is utterly broken".

**Comment by Xhuis** [comment]
And we were doing so well with keeping things civil up until now.

**Comment by AMurkin** [comment]
> But I have cleared out multiple hospitals, and the amount of anesthetic I've found? None.

All the progress of the character occurred after the introduction of anesthesia sets? Or is it a regret for the lost opportunity to install all the modules because of the update?

**Comment by Neon1ight** [comment]
@Night-Pryanik I understand that you're personally invested in the system, but I _am_ giving constructive feedback. Anesthetic is so rare to the point of rendering CBMs pointless-- I am describing this point in the form of an anecdote. I never at any point claimed anesthetic kits should be common-- But you would think that you'd be able to find at least a few in a hospital, somewhere where surgical operations happen often. I have been following development of this game for years now and I have never claimed to think that issues with balancing and development were easily solved in a singular PR-- And my point stands regardless. Claiming a system that is woefully ill-balanced is utterly broken is not shitty whining. It is my interpretation of the system and it's current state. Thank you for addressing my misunderstanding of what autodocs are. I ask that in the future that you be more respectful in the act of responding to someone's input-- The behavior you've exhibited in doing so (Ad hominem attacks) is something that turns people off from contributing to the project.

**Comment by Xhuis** [comment]
@Neon1ight As much as I generally try to stay out of arguments, you somewhat provoked the argument yourself with your extreme reaction to the problem and attacks against his work, such as:
> This system is utterly broken, and it makes it so bionics are so expensive that the system is utterly cost-inefficient to the point of lunacy. They are completely pointless, at this stage.

This is not constructive feedback. This is anger.

**Comment by Neon1ight** [comment]
Not my intention to provoke anyone, so I'll apologize on that front.

**Comment by Night-Pryanik** [comment]
> But you would think that you'd be able to find at least a few in a hospital, somewhere where surgical operations happen often.

Look, it's just RNG acting bad. I added a chance anesthesia kits to spawn in a hospital (inside a `surgery` item group). There are several places where items from this group should spawn, so I thought that even with relatively (!) low chance to spawn of single anesthesia kit, overall it may be enough. Looks like it was too low nevertheless.

I just want to say THIS is constructive criticism, like "Hospital looks like a good place to find rather big amounts of anesthesia, but I couldn't find not a single one in a several hospitals a row." This is a good rationale for increasing spawn chance of maybe even give it a guaranteed spawn.

**Comment by Night-Pryanik** [comment]
> But you would think that you'd be able to find at least a few in a hospital, somewhere where surgical operations happen often.

Look, it's just RNG acting bad. I added a chance anesthesia kits to spawn in a hospital (inside a `surgery` item group). There are several places where items from this group should spawn, so I thought that even with relatively (!) low chance to spawn of single anesthesia kit, overall it may be enough. Looks like it was too low nevertheless.

I just want to say THIS is constructive criticism, like "Hospital looks like a good place to find rather big amounts of anesthesia, but I couldn't find not a single one in a several hospitals a row." This is a good rationale for increasing spawn chance of maybe even give it a guaranteed spawn.

**Comment by l29ah** [comment]
It may make sense to be able to install the power storage CBMs using the old method utilizing the painkillers, what do you think?

**Comment by kevingranade** [comment]
What makes you think that?

**Comment by l29ah** [comment]
Batteries don't sensibly interact with any tissue and don't require augmentation of any organs, and even today's implant lovers don't have problems with implanting dumb stuff into their bodies themselves even without painkillers, using scalpel, sewing needle and local anesthesia.

**Comment by kevingranade** [comment]
Overall issues addressed, if you want to pick out specific elements brought up that are yet to be implemented, feel free to open PRs about them.
